src/decode.rs

```diff
@@ -743,7 +830,16 @@ impl<C: CodeBuffer> Stateful for DecodeState<C> {
         let mut last_decoded: Option<&[u8]> = None;
 
         while let Some((mut code, mut link)) = code_link.take() {
-            if out.is_empty() && !self.buffer.buffer().is_empty() {
+            let want_more_space = {
+                // We have more data buffered but not enough space to put it? We want fetch a next
+                // symbol if possible as in the case of it being a new symbol we can refer to the
+                // buffered output as the source for that symbol's meaning and do a memcpy. It
+                // would be correct to break here instead but then that symbol must be
+                // reconstructed by iterating the code table.
+                out.is_empty()
+            };
```

Comment:
You'll note that this condition is not equivalent to the prior code even with the flag being false. I tried that at first but for some reason it confuses the optimizer.. There was a pretty consistent 8% performance hit with the benchmarks even with a const condition if involved. It's.. weird. Could be a measurement artifact but I doubt it.